# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: editor/libeditor/crashtests/639736-1.xhtml
# Commit: 9b15baab6f77
# Full Hash: 9b15baab6f77d09eca2956276f66fee97555e4a1
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2022-09-22 09:56:47
# Regressor Bug: 1789967
# File Overlap Count: 1
# Description:
#   Bug 1789967 - part 3: Make `HTMLEditor::CollapseSelectionToEndOfLastLeafNodeOfDocument` and `HTMLEditor::InitEditorContentAndSelection` do nothing if the document is partially editable r=m_kato
#   
#   They and their callees work with the result of `GetRoot()` which is the document
#   element or the body element.  If the body is not editable, `Selection` should
#   not be updated in non-editable region nor `<br>` elements should not be
# ==============================================================================

diff -r 4d1e93c629da -r 9b15baab6f77 editor/libeditor/crashtests/639736-1.xhtml
--- a/editor/libeditor/crashtests/639736-1.xhtml	Thu Sep 22 06:24:38 2022 +0000
+++ b/editor/libeditor/crashtests/639736-1.xhtml	Thu Sep 22 06:27:37 2022 +0000
@@ -1,14 +1,19 @@
 <html xmlns="http://www.w3.org/1999/xhtml">
 <head>
 <script>
-
-function boom()
-{
-  try { document.execCommand("removeformat", false, null); } catch(e) { }
+function onLoad() {
+  // For emulating the traditional behavior, collapse Selection to end of the
+  // <td> which is the deepest last child of the <body>.
+  getSelection().collapse(
+    document.querySelector("td"),
+    document.querySelector("td").childNodes.length
+  );
+  try {
+    document.execCommand("removeformat");
+  } catch(e) {}
   document.adoptNode(document.documentElement);
 }
-
 </script>
 </head>
-<body onload="boom();"><td contenteditable="true" /></body>
+<body onload="onLoad();"><td contenteditable="true"/></body>
 </html>